Examples.md

```diff
@@ -6,17 +6,41 @@ how to do this in various languages.
 
 ## C
 
-```c
+```C
 #include <stdio.h>
 #include <stdlib.h>
 
-int main(void) {
-    printf("Hello, World!\n");
+int main(void)
+{
+        // stdout buffer setting, Python way
+        // setvbuf(stdout, (char *)NULL, _IONBF, BUFSIZ);
+        // OR
+        // stdout buffer setting, Perl way
+        setvbuf(stdout, NULL, _IOLBF, 0);
 
-    if (fflush(stdout) != 0 || ferror(stdout) != 0) {
-        return EXIT_FAILURE;
-    }
+        printf("Hello, World!\n");
+        if (ferror(stdout) || feof(stdout)) {
+                perror("ERROR: printing failed");
+                return EXIT_FAILURE;
+        }
 
-    return EXIT_SUCCESS;
+        return fflush(stdout);
+}
+```
+
+## C++
+
+```C++
+#include <iostream>
+#include <cstdlib>
+using namespace std;
+
+int main(void)
+{
+        if (!(cout << "Hello, World!" << endl)) {
```

Comment:
Yes :) it can be:
```C++
#include <iostream>
#include <cstdlib>
using namespace std;

int main(void)
{
        auto &str = cout << "Hello, World!" << endl;
        if (!str) {
                perror("ERROR: printing failed");
                return EXIT_FAILURE;
        }
        return EXIT_SUCCESS;
}
```

Reply:
I'd like the C++ example to do the `cout << ...` parts without capturing the return value, and to check the error state stored in `cout` in a separate statement. The reason is that I expect this more closely reflects how people actually use `cout` in practice.